src/components/ContentHighlights/ContentHighlightSetCard.jsx

```diff
@@ -1,23 +1,48 @@
-import React from 'react';
-import { Card } from '@edx/paragon';
+import React, {
+  useMemo, useContext,
+} from 'react';
+import { Card, Badge, Stack } from '@edx/paragon';
 import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
 import { useHistory } from 'react-router-dom';
 import { ROUTE_NAMES } from '../EnterpriseApp/constants';
+import { HIGHLIGHT_CARD_BADGE_STATUS } from './data/constants';
+import { ContentHighlightsContext } from './ContentHighlightsContext';
 
 const ContentHighlightSetCard = ({
-  title, highlightUUID, enterpriseSlug,
+  title, highlightUUID, isPublished, enterpriseSlug,
 }) => {
   const history = useHistory();
+  /* Stepper Draft Logic (See Hook) - Start */
+  const {
+    setIsModalOpen,
+  } = useContext(ContentHighlightsContext);
+  /* Stepper Draft Logic (See Hook) - End */
+  const handleHighlightSetClick = () => {
+    if (isPublished) {
+      // redirect to individual highlighted courses based on uuid
+      return history.push(`/${enterpriseSlug}/admin/${ROUTE_NAMES.contentHighlights}/${highlightUUID}`);
+    }
+    return setIsModalOpen(true);
+  };
+  const badgeData = useMemo(() => () => ({
```

Comment:
[nit/suggestion] Return the object, not a function: `useMemo(() => ({ ... })`, so you don't need to call the function when you need to read `badgeData`